mobile/native/src/db/models.rs

```diff
@@ -460,6 +493,33 @@ pub enum OrderState {
     Filled,
 }
 
+impl OrderState {
+    /// Determines what state to go to after learning about the latest [`OrderState`] update.
+    ///
+    /// If the state should remain unchanged [`None`] is returned.
+    ///
+    /// TODO: It might be a good idea to introduce a different type that models `OrderUpdates`
+    /// explicitly.
+    fn next_state(&self, latest: Self) -> Option<Self> {
```

Comment:
What do you think of adding some tests ensuring that this never changes and giving some context about the state transitions?